static/js/controllers/reports-content.js

```diff
@@ -45,15 +47,45 @@
         setMessageState(group, 'muted', 'scheduled');
       };
 
+      $scope.toggleExpand = function(selection) {
+        if ($scope.selectMode) {
+          if (selection.report || selection.expanded) {
+            selection.expanded = !selection.expanded;
+          } else {
```

Comment:
The search performance changes introduced a `summary` property which has all the data needed for the collapsed version. The `report` is only populated on first expand but then it's cached for future expansions.

The `else` is all about loading the `report` property, but if the `report` property is already populated or we're collapsing the row (eg: `selection.expanded === true`) then don't worry about loading the report.